src/pages/Integrations/List/List.tsx

```diff
@@ -278,7 +295,14 @@ const IntegrationsList: React.FunctionComponent<IntegrationListProps> = ({
           onClose={closeFormModal}
         />
       )}
-      {wizardEnabled && category && (
+      {modalIsOpenState.isTest && (
+        <IntegrationTestProvider
+          integrationUUID={selectedIntegrationID}
+          onClose={() => setIsTestModalOpen(false)}
+          isModalOpen={isTestModalOpen}
+        />
+      )}
+      {wizardEnabled && category && !modalIsOpenState.isTest && (
```

Comment:
I'm not a huge fan of making these conditional rendering blocks even more complex than they already are - they do not make it abundantly clear what will render and why (making it harder and harder to diagnose if there is an issue). Also what happens if the delete or create modal are already open in this situation?